client/cache/write_delete.go

```diff
@@ -47,3 +49,39 @@ func (w *WriteDeleteCache) Set(ctx context.Context, key string, val any) error {
 	}
 	return w.Cache.Delete(ctx, key)
 }
+
+// WriteDoubleDeleteCache creates write double delete cache pattern decorator.
```

Comment:
it will delete the key from cache when you call Set function, and wait for `interval`, it will delete the key from cache one more time.
This pattern help to reduce the possibility of data inconsistencies, but it's still possible to be inconsistent among database and cache.